serenata_toolbox/datasets/__init__.py

```diff
@@ -0,0 +1,73 @@
+import os
+
+from serenata_toolbox.datasets.downloader import Downloader
+from serenata_toolbox.datasets.local import LocalDatasets
+from serenata_toolbox.datasets.remote import RemoteDatasets
+
+
+class Datasets:
+    """
+    This is a wrapper for three different classes together handles the datasets
+    locally and remotelly.
+
+    Inside the instantiated object there are three main objects: local, remote,
+    and downloader:
+
+    * `Datasets.local` handles listing all local datasets through the property
+      `Datasets.local.all` and deleting local datasets with the method
+      `Datasets.local.delete(filename)`;
+
+    * `Datasets.remote` has the `Datasets.remote.all` property and
+      `Dataset.remote.delete(filename)` method just like its local equivalent;
+      in addition to them this object offers the
+      `Datasets.remote.upload(file_path)` method to upload a local file to the
+      remote bucket; `Datasets.remote` does not handles downloads because
+      `boto3` does not support `asyncio` and we prefer to use async tasks to
+      allow the download of more than one file in parallel;
+
+    * `Datasets.downloader` implements a async manager to download files from
+      the remote bucket. It's `Datasets.downloader.download(files)` take the
+      path for a single file (str) as argument or an iterable of paths (str).
+
+    Yet this wrapper implement the `Dataset.upload_all()` method to upload all
+    local datasets that are not present in the remote bucket.
+    """
+
+    def __init__(self, local_directory=None):
+        """
+        :param local_directory: (str) path to local directory of the datasets
+        """
+        if not local_directory:
+            local_directory = 'data'
```

Comment:
I prefer to leave it as `serenata-de-amor` default, not as `rosie` repo default. Within Rosie we kind of have full control of when and how `Datasets` will be instaciated but within Serenata it can happen in different parts (`src/`, notebooks, `setup` script etc.).